Add the workshop 2022 program (#124)
precice/precice.github.io: pages/community/precice-workshop-2022.md

## Program

The workshop stretches from Monday noon to Thursday evening.

### Monday, February 21

* 12:30 - 13:00 Welcoming (Main Hall)
* 13:00 - 14:30 Developer talks

>     <details class="workshop-event" id="Simonis and Chourdakis"><summary>Frédéric Simonis and Gerasimos Chourdakis: Whats new in preCICE?</summary>

Reviewer: ```suggestion
    <details class="workshop-event" id="Simonis and Chourdakis"><summary>Frédéric Simonis and Gerasimos Chourdakis: What's new in preCICE?</summary>
```

    <p>Authors: <a>Frédéric Simonis</a>, <a>Gerasimos Chourdakis</a> <br/></p>
    <p>The core library became more robust, user-, and resource-friendly, while we restructured the tutorials and polished both bindings and adapters.
    In this talk, we will explore released as well as upcoming changes to both the library and the ecosystem as a whole.</p>
    </details>

    <details class="workshop-event" id="Uekermann"><summary>Benjamin Uekermann: You support preCICE – preCICE supports you</summary>
    <p>Authors: <a>Benjamin Uekermann</a> <br/></p>

>     <p>Funding open-source research software such as preCICE in a sustainable manner is a challenge. Moreover, during the last year, the amount of support requests and collaborations has reached the manageable limit of the preCICE core team. We are introducing a preCICE support program to solve both issues – such that you can support preCICE and, thus, we can support you in a better and more reliable way. In this talk, we step you through our thought process that led to the program and discuss its funding and support options.</p>

Reviewer: @uekerman a comment I once got from the English Writing Center is that "last year" sounds a bit like "there will be no other year". https://ell.stackexchange.com/a/160767/134614

```suggestion
    <p>Funding open-source research software such as preCICE in a sustainable manner is a challenge. Moreover, during the past year, the amount of support requests and collaborations has reached the manageable limit of the preCICE core team. We are introducing a preCICE support program to solve both issues – such that you can support preCICE and, thus, we can support you in a better and more reliable way. In this talk, we step you through our thought process that led to the program and discuss its funding and support options.</p>
```

    </details>

> * 14:30 - 15:00 Speakers in break out rooms for Q&A

Reviewer: I would suggest the form `breakout`, as used by [Zoom](https://support.zoom.us/hc/en-us/articles/206476313-Managing-Breakout-Rooms) and [BBB](https://support.bigbluebutton.org/hc/en-us/articles/1500005216981-Create-and-manage-breakout-rooms).

```suggestion
* 14:30 - 15:00 Speakers in breakout rooms for Q&A
```

* 15:00 - 15:30 Break (Offline)
* 15:30 - 17:00 preCICE Course I (Main Hall)

>   <details class="workshop-event" id="courseI"><summary>preCICE Course I: Basics and Tools</summary>

Reviewer: ```suggestion
  <details class="workshop-event" id="courseI"><summary>preCICE Course I: Basics</summary>
```

  <p>Instructors: <a href="https://www.in.tum.de/en/i05/people/personen/gerasimos-chourdakis/">Gerasimos Chourdakis</a>, <a href="https://www.simtech.uni-stuttgart.de/exc/people/Uekermann/">Benjamin Uekermann</a><br/>
  Affiliation: Technical University of Munich, University of Stuttgart, preCICE developers.</p>
  <p>A hands-on introduction to preCICE, recommended for new users that want to learn how to couple their own codes.</p>

>   <p>We will couple two simple Python codes, discussing the basic methods of the preCICE API and the structure of the configuration file. We will then also look into tools useful for developing and debugging coupled simulations.</p>

Reviewer: This part needs to go to day 2:

```suggestion
  <p>We will couple two simple Python codes, discussing the basic methods of the preCICE API and the structure of the configuration file.</p>
```

Author: This will also change the requirements, right? Can you comment on that? I pushed this part already, but are not completely done with your review.

Reviewer: The technical requiremenets for pt2 are the same as for pt1.

>   <p>We will need preCICE v2.3.0, Python 3.6 or newer, and the Python bindings. Optionally, please also install ParaView and gnuplot, or similar software to visualize VTK point data and CSV files. If you prefer to try everything in a Virtual Machine, you can use the <a href="https://app.vagrantup.com/precice/boxes/precice-vm">Vagrant Box</a> (VirtualBox image) <a href="https://app.vagrantup.com/precice/boxes/precice-vm">precice/precice-vm</a>.</p>

Reviewer: ```suggestion
  <p>We will need preCICE v2.3.0, Python 3.6 or newer, and the Python bindings. Optionally, please also install ParaView and gnuplot, or similar software to visualize VTK point data and CSV files. If you prefer to try everything in a Virtual Machine before setting up your own system (recommended), you can use the <a href="https://precice.org/installation-vm.html">preCICE Demo Virtual Machine</a>.</p>
```

  <p>See the "preCICE Course II" for more topics and additional technical requirements.</p>
  </details>

### Tuesday, February 22

* 09:45 - 10:00 Morning synchronization (Main Hall)
* 10:00 - 12:00 User presentations (Main Hall)<br/>
  See preCICE in action, in a variety of applications. Each talk will be an approx. 20min video premiere, followed by a live Q&A session.
  Click on each box for more details.<br/>

    <details class="workshop-event" id="Saha"><summary>Sanjoy Kumar Saha: Aero‐Structural Analysis of Flexible Structure Membrane Aeroshell</summary>

>     <p>Authors: <a>Sanjoy Kumar Saha</a><br/></p>

Reviewer: It would be very interesting to have the affiliation next to the authors.

Author: Added it

  Affiliation: Technical University of Munich, University of Stuttgart, preCICE developers.</p>
  <p>A hands-on introduction to implicit coupling details in preCICE, recommended for new users that want to learn how to make their coupled simulations more accurate and numerically efficient.</p>
  <p>We will couple OpenFOAM and Nutils for Conjugate Heat Transfer, discussing the basic methods of the preCICE API and the structure of the configuration file. We will then also look into tools useful for developing and debugging coupled simulations.</p>
  <p>We will need preCICE v2.3, Python 3.6 or newer, the Python bindings, Nutils 6, OpenFOAM (e.g. v2112), the latest OpenFOAM adapter, and ParaView. If you prefer to try everything in a Virtual Machine, you can use the <a href="https://app.vagrantup.com/precice/boxes/precice-vm">Vagrant Box</a> (VirtualBox image) <a href="https://app.vagrantup.com/precice/boxes/precice-vm">precice/precice-vm</a>.</p>
  <p>If you are new to preCICE, see "preCICE Course I" first. If you already understand the basics of an adapter, you can directly start here.</p>

>   </details>

Reviewer: - The whole thing should go to Day 3
- Same change for the VM
- "preCICE Course I and II first" (or remove)
- The updates for versions look good, thanks!

* 16:30 - 17:00 Break (Offline)

> * 17:25 - 18:15 Networking/Social Event (Main Hall)<br/>

Reviewer: What is the motivation for cutting it already at 18:15? Gather.Town limits?

Author: @BenjaminRodenberg do we have any limitations here?

Reviewer 2: We have two options:
* Pay for two hours
* Pay for a day

If we pay for a day then we can use Gather.Town that whole day and also have an open-ended social event. I am tending towards paying for the day because some people might want to hang around for a longer time. Opinions?

Reviewer 2: Lets have the social event for two hours for now.

  Get to know each other and discuss your work in more detail, or simply enjoy a relaxed atmosphere after the user presentation talks.

### Wednesday, February 23

* 09:45 - 10:00 Morning synchronization (Main Hall)
* 10:00 - 12:00 Developer talks (Main Hall)<br/>
  Learn about the latest developments or other important elements of preCICE.
  Similarly to the speaker presentations, the talks will be pre-recorded, followed by a live Q&A session.
  Click on each box for more details.<br/>

   <details class="workshop-event" id="Desai"><summary>Ishaan Desai: Adaptive and flexible macro micro coupling software</summary>

>    <p>Authors: <a>Ishaan Desai</a> <br/></p>

Reviewer: Let's link to every speaker's university page we know, so that people have at least contact details.

Alternatively, we could link to the Discourse introduction posts. (me and BU are already linked in the course, but nobody else is)

Author: I added links to all of our talks, but not to the user talks since I don't have every page.

   <details class="workshop-event" id="Schneider"><summary>David Schneider: Data Mapping in preCICE</summary>
   <p>Authors: <a>David Schneider</a> <br/></p>
   <p>This talk provides an overview of current mesh mapping methods in preCICE. A more detailed explanation of the radial basis function implementation is provided, along with some mesh mapping test results. Throughout the talk, best practice methods are suggested, and rounded off with future plans to enable easier mapping in preCICE.</p>
   </details>

* 12:00 - 12:30 Speakers in breakout rooms for Q&A
* 12:30 - 13:30 Break (Offline)
* 13:30 - 15:00 preCICE Course III (Main Hall)

  <details class="workshop-event" id="courseIII"><summary>preCICE Course III: </summary>
  <p>Instructors: <a href="https://www.in.tum.de/en/i05/people/personen/gerasimos-chourdakis/">Gerasimos Chourdakis</a>, <a href="https://www.simtech.uni-stuttgart.de/exc/people/Uekermann/">Benjamin Uekermann</a><br/>
  Affiliation: Technical University of Munich, University of Stuttgart, preCICE developers.</p>
  </details>

* 14:30 - 15:00 Speakers in breakout rooms for Q&A
* 15:00 - 16:00 Break (Offline)
* 16:00 - 17:00 Workshop feedback (Main Hall)

>   * The World Cafe-styled feedback session will provide the chance for participants to give feedback on the virtual workshop, the usability of preCICE and the community in general.

Reviewer: ```suggestion
  * The World Café-styled feedback session will provide the chance for participants to give feedback on the virtual workshop, the usability of preCICE, the support program, and the community in general.
```

(here we still need some updates, I guess)

>   * Participants will be divided into breakout rooms, where a moderator will present a shared pad and motivate the discussion.

Reviewer: Does this really need to be a separate bullet point?

    </details>

   <details class="workshop-event" id="Schneider"><summary>David Schneider: Data Mapping in preCICE</summary>
   <p>Authors: <a>David Schneider</a> <br/></p>
   <p>This talk provides an overview of current mesh mapping methods in preCICE. A more detailed explanation of the radial basis function implementation is provided, along with some mesh mapping test results. Throughout the talk, best practice methods are suggested, and rounded off with future plans to enable easier mapping in preCICE.</p>
   </details>

* 12:00 - 12:30 Speakers in breakout rooms for Q&A
* 12:30 - 13:30 Break (Offline)
* 13:30 - 15:00 preCICE Course III (Main Hall)

  <details class="workshop-event" id="courseIII"><summary>preCICE Course III: </summary>
  <p>Instructors: <a href="https://www.in.tum.de/en/i05/people/personen/gerasimos-chourdakis/">Gerasimos Chourdakis</a>, <a href="https://www.simtech.uni-stuttgart.de/exc/people/Uekermann/">Benjamin Uekermann</a><br/>
  Affiliation: Technical University of Munich, University of Stuttgart, preCICE developers.</p>

>   </details>

Author: @MakisH @uekerman I need some input on how you want to split the course content here in order to provide a proper summary in the program.

Reviewer: Done

## Program

> The workshop stretches from Monday noon to Thursday evening.

Reviewer 2: Lets mention the timezone here and also all the other places where we mention time. So write CET

### Monday, February 21

* 12:30 - 13:00 Welcoming (Main Hall)

> * 13:00 - 14:30 Developer talks

Reviewer 2: Time zone is necessary

    <details class="workshop-event" id="Simonis and Chourdakis"><summary>Frédéric Simonis and Gerasimos Chourdakis: What's new in preCICE?</summary>

>     <p>Authors: <a href="https://www.in.tum.de/en/i05/people/personen/frederic-simonis/">Frédéric Simonis</a>, <a href="https://www.in.tum.de/en/i05/people/personen/gerasimos-chourdakis/">Gerasimos Chourdakis</a><br/>

Reviewer 3: Can you swap the two names?
Chourdakis should come before Simonis.
